core/src/subgraph/instance_manager.rs

```diff
@@ -663,7 +688,12 @@ where
     .await
     {
         Ok(block_state) => block_state,
-        Err(MappingError::Unknown(e)) => return Err(e.compat_err().into()),
+        Err(MappingError::Unknown(e)) => {
+            return Err(CancelableError::Error(ProcessBlockError::Unknown(e)))
```

Comment:
Is there any advantage to have separate `MappingError` and `ProcessBlockError`? Maybe we should just elevate `MappingError` to be used throughout?

Reply:
`ProcessBlockError` doesn't have the `PossibleReorg` variant.

Reply:
But for that one, we know that it's not deterministic for sure.

Reply:
I just see it as an advantage to have a fine grained types with less variants, and I don't see what code would be greatly simplified by having a single type.